panic when running tilt logs

From https://github.com/tilt-dev/tilt/issues/4451#issuecomment-823487435 - you can get index out of range errors with `tilt logs`:
```sh
$ tilt logs -f "resource"
panic: runtime error: index out of range [-1]

goroutine 138 [running]:
github.com/tilt-dev/tilt/pkg/model/logstore.(*LogStore).prevIndexMatchingManifests(...)
        /root/project/pkg/model/logstore/logstore.go:203
github.com/tilt-dev/tilt/pkg/model/logstore.(*LogStore).ContinuingLinesWithOptions(0xc0001bea00, 0x517, 0xc000ba4ab0, 0xc000c51801, 0xc0010e2706, 0x7, 0xc017e2fffb818540)
        /root/project/pkg/model/logstore/logstore.go:451 +0x9cd
github.com/tilt-dev/tilt/internal/hud/server.(*LogStreamer).Handle(0xc0003763a0, 0x0, 0x0, 0xc000572100, 0x16, 0x20, 0x0, 0xc0010d66c0, 0x0, 0xc0010e40c0, ...)
        /root/project/internal/hud/server/logs_reader.go:97 +0x234
github.com/tilt-dev/tilt/internal/hud/server.(*WebsocketReader).handleTextMessage(0xc0004ae9a0, 0x678c628, 0xc00004f340, 0x103ada78, 0xc000c07600, 0x0, 0x0)
        /root/project/internal/hud/server/logs_reader.go:168 +0x13b
github.com/tilt-dev/tilt/internal/hud/server.(*WebsocketReader).Listen.func1(0xc0000c52c0, 0xc0004ae9a0, 0x678c628, 0xc00004f340)
        /root/project/internal/hud/server/logs_reader.go:134 +0xc5
created by github.com/tilt-dev/tilt/internal/hud/server.(*WebsocketReader).Listen
        /root/project/internal/hud/server/logs_reader.go:125 +0x91
```

--- a/pkg/model/logstore/logstore.go
+++ b/pkg/model/logstore/logstore.go
@@ -199,7 +199,7 @@ func (s *LogStore) prevIndexMatchingManifests(index int, mns model.ManifestNameS
 		return index - 1
 	}
 
-	for i := index - 1; index >= 0; i-- {
+	for i := index - 1; i >= 0; i-- {
 		span, ok := s.spans[s.segments[i].SpanID]
 		if !ok {
 			continue

